### PR TITLE
Server setup

### PR DESCRIPTION
**Pull request by alishaedington**
any input would be appreciated
if you see any holes, please let me know.

### CODE REVIEW

File: config.example.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const pgPwd = 'FILL_ME_IN';
+
+module.exports = pgPwd;
```
Comment (adbarc92) [inline config.example.js]: This could be done in one line, unless you foresee more exports later

File: database/index.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  host: 'localhost',
+  database: 'reviews',
+  password: pgPwd,
+  port: 5432,
+});
+
+// connection.connect((err) => {
+//   if (err) { console.log(err); }
+//   else { console.log('connected to DB'); }
+// });
```
Comment (adbarc92) [inline database/index.js]: Commented code should be removed if it isn't being saved for future use

File: package.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  "author": "Alisha Edington",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/alex-richard-alisha/sdc-ratings-and-reviews/issues"
+  },
+  "homepage": "https://github.com/alex-richard-alisha/sdc-ratings-and-reviews#readme",
+  "dependencies": {
+    "cors": "^2.8.5",
+    "express": "^4.17.1",
+    "nodemon": "^2.0.12",
```
Comment (adbarc92) [inline package.json]: `nodemon` should be a dev dependency

File: server/controllers.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+const getReviews = async (request, response) => {
+  const { product_id } = request.query;
+  console.log('productId', product_id);
+  const client = await pool.connect();
+
+  try {
+    const res = await client.query('SELECT reviews.id, reviews.rating, reviews.summary, reviews.recommend, reviews.response, reviews.body, reviews.date, reviews.reviewer_name, reviews.helpfulness, ARRAY(select row (review_photos.id, review_photos.url) from review_photos where review_photos.review_id = reviews.id) as photos from reviews where product_id = $1;', [product_id]);
+    console.log('result from query', res);
+    response.send(res);
```
Comment (adbarc92) [inline server/controllers.js]: Missing an error catch for the `try`

File: server/controllers.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    client.release();
+  }
+};
+
+const getReviewsMeta = async (request) => {
+  const client = await pool.connect();
+
+  try {
+    const res = client.query('some cool sql stuff', [])
+    console.log(res.rows[0]);
```
Comment (adbarc92) [inline server/controllers.js]: Same as above

File: server/routes.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+const router = require('express').Router();
```
Comment (adbarc92) [inline server/routes.js:1]: Is there a particular reason why you're using a router here?

**Review by adbarc92 — CHANGES_REQUESTED** [review]
A few small changes suggested, along with a question. Things look good, overall.

File: config.example.js (hunk 1 above)
Comment (adbarc92) [reply config.example.js]: `module.exports = 'FILL_ME_IN'`, or an object if you're intending to add other config options in the future.

Comment (alishaedington) [reply config.example.js]: okokok i see you

File: server/routes.js (hunk 6 above)
Comment (alishaedington) [reply server/routes.js:1]: I'm defining the routes on the router here and exporting them.

File: server/controllers.js (hunk 4 above)
Comment (alishaedington) [reply server/controllers.js]: gotcha

File: server/controllers.js (hunk 5 above)
Comment (alishaedington) [reply server/controllers.js]: doneso!

File: package.json (hunk 3 above)
Comment (alishaedington) [reply package.json]: fixed.

File: database/index.js (hunk 2 above)
Comment (alishaedington) [reply database/index.js]: deleted!

File: database/reviewsSchema.sql [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+COPY review_photos FROM 'Users/alishaedington/work/SDC/SDC_files/review_photos.csv' delimiter ',' csv header;
+
+CREATE TABLE characteristics (
+  id INTEGER PRIMARY KEY,
+  product_id INTEGER NOT NULL,
+  name TEXT
+);
+
+COPY characteristics FROM '.Users/alishaedington/work/SDC/SDC_files/characteristics.csv' delimiter ',' csv header;
```
Comment (adbarc92) [inline database/reviewsSchema.sql]: You could make path changes to this now, or wait until it actually matters again. I'd probably make a note of it.

File: server/controllers.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+const services = require('./services');
+const pool = require('../database/index');
+
+const getReviews = async (request, response) => {
+  const { product_id } = request.query;
+  const client = await pool.connect();
+
+  try {
+    const res = await client.query('SELECT reviews.id, reviews.rating, reviews.summary, reviews.recommend, reviews.response, reviews.body, reviews.date, reviews.reviewer_name, reviews.helpfulness, (select json_agg(photos) from (select review_photos.id, review_photos.url from review_photos where review_photos.review_id = reviews.id) as photos) from reviews where product_id = $1;', [product_id]);
```
Comment (adbarc92) [inline server/controllers.js]: Does this work without the parens around the query variable? `($1)`

**Review by adbarc92 — APPROVED** [review]

**Pull request merged**